src/janitor.c

```diff
@@ -595,6 +596,38 @@ static void pool_server_maint(PgPool *pool)
 		}
 	}
 
+	/* handle user level idle_transaction_timeout */
+	if (any_user_level_timeout_set) {
+		statlist_for_each_safe(item, &pool->active_server_list, tmp) {
```

Comment:
I think this loop should be combined with the loop above. The user level timeout should override the global one, so if the user has a higher timeout than the global one, the connection should not disconnect when the global one expires, only when the user one expires. Right now this isn't the case afaict.

Reply:
Agreed. Will modify accordingly.

Reply:
Combined the loops as suggested with the override logic. Though it would probably make sense to unit test that user level timeouts correctly overwrite the global level timeouts.